### PR TITLE
AbstractMapDataset: add option for mapset check

### PR DESCRIPTION
**Pull request by ninsbl**
Currently, `t.register` and the `register` module use the `build_id` function from `AbstractMapDataset`, but at the same time duplicate some of the code from that function. This PR takes the additional functionality the `register` module attaches to the `build_id()` function and moves it into the `AbstractMapDataset.build_id()` function. Then duplicated code can be removed from the register module (see #2863).

The behavior of the `build_id` function remains as it is. So places where it is currently used are not affected by the proposed change. Just one "if-check" is added in those cases.

This PR is a precondition for #2863.

### CODE REVIEW

**Comment by ninsbl** [comment]
This would be ready to merge, along with https://github.com/OSGeo/grass/pull/2863 
Any opinions on the way the issue with the `build_id` function is tackled in those PRs (see #2863 for discussion)?
OK to merge?

File: python/grass/temporal/abstract_map_dataset.py [hunk 1]
```diff
@@ -165,33 +166,52 @@ def get_map_id(self):
...
         """Convenient method to build the unique identifier
 
         Existing layer and mapset definitions in the name
         string will be reused
 
+        If check_mapset_element is given, but no mapset, the function will
+        try to get the correct mapset by looking for an element of the
+        given element type with the given name. If the combination is
+        not found on the current search path, it will fail and throw an error.
+
```
Comment (petrasovaa) [inline python/grass/temporal/abstract_map_dataset.py:179]: Perhaps add that check is not executed if mapset is given. Given the current parameter name `check`, this behavior seems unexpected to me. Would it make sense to check it in any case?

Comment (petrasovaa) [reply python/grass/temporal/abstract_map_dataset.py:179]: Or, change the parameter name to `element`.

**Comment by wenzeslaus** [comment]
> ...Any opinions on the way the issue with the `build_id` function is tackled in those PRs (see #2863 for discussion)?...

The new build_id does two or three things here, retrieving mapset, enforcing existence at user-level (gs.fatal is ending module execution by default), and building a string id from provided information. It makes it really hard to guess what the function will do when reading the caller code.

You need at least two functions to make it clear: build_id (the original) and build_id_for_existing_map which is more expected to throw some kind of error when map does not exist. Looking my old comment (https://github.com/OSGeo/grass/pull/2863#discussion_r1125224648), I though back then you may need a third function which does the parsing like split_name_layer_mapset and is used in both functions, i.e., build_id does some splitting now which would be separated out. I was also suggesting ensure_mapset, but the build_id_for_existing_map route seems closer to what you have now.

Separate build_id, split_... make it easy to test these functions separately, but also just as simple string manipulation functions without a need to get involved with actual data.

**Comment by ninsbl** [comment]
@wenzeslaus is [cb41621](https://github.com/OSGeo/grass/pull/2882/commits/cb41621cf37e680e2b557ecf7bbd692844165978) along the lines you thought?

**Comment by wenzeslaus** [comment]
> @wenzeslaus is [cb41621](https://github.com/OSGeo/grass/pull/2882/commits/cb41621cf37e680e2b557ecf7bbd692844165978) along the lines you thought?

Yes!

**Comment by ninsbl** [comment]
OK, then I take the liberty to merge this and next include `build_id_from_search_path` in _register.py_.

**Pull request merged**